src/Illuminate/Cache/Repository.php

```diff
@@ -624,7 +624,9 @@ protected function getSeconds($ttl)
         $duration = $this->parseDateInterval($ttl);
 
         if ($duration instanceof DateTimeInterface) {
-            $duration = Carbon::now()->diffInSeconds($duration, false);
+            $duration = (int) ceil(
+                Carbon::now()->diffInMilliseconds($duration, false) / 1000
```

Comment:
behavior will be slightly variable based on whether or not application is using Carbon v2 or v3, right?

v2 returns an integer, and v3 returns a float. is this why you switched from `diffInSeconds()` to `diffInMilliseconds()` to decrease that variability?

Reply:
hah, your comment answers my question.

Reply:
Didn't see you question, sorry. Hopefully that makes sense. Let me know any concerns.